Anixe.QualityTools/AssertHelper.cs

```diff
@@ -1,154 +1,169 @@
+using FluentAssertions;
+using FluentAssertions.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using System;
 using System.Collections.Generic;
 using System.Text;
 using System.Xml.Linq;
-using FluentAssertions;
-using FluentAssertions.Json;
-using Newtonsoft.Json.Linq;
-using Newtonsoft.Json;
 using Xunit;
 
 namespace Anixe.QualityTools
-{
-  public static class AssertHelper
-    {
-        public static void AreXmlDocumentsEqual(string expected, string actual)
-        {
-            var actualXml = XDocument.Parse(actual);
-            var expectedXml = XDocument.Parse(expected);
-            try
-            {
-                actualXml.Should().BeEquivalentTo(expectedXml);
-            }
-            catch (Xunit.Sdk.XunitException ex)
+{
+  public static class AssertHelper
+  {
+    public static void AreXmlDocumentsEqual(string expected, string actual)
+    {
+      var actualXml = XDocument.Parse(actual);
+      var expectedXml = XDocument.Parse(expected);
+      try
+      {
+        actualXml.Should().BeEquivalentTo(expectedXml);
+      }
+      catch (Xunit.Sdk.XunitException ex)
+      {
+        var expectedMsg = expectedXml.ToString();
+        var actualMsg = actualXml.ToString();
+
+        var sb = new StringBuilder(expectedMsg.Length + actualMsg.Length + ex.Message.Length + 70)
+          .AppendLine(ex.Message)
+          .AppendLine()
+          .AppendLine("################### expected:")
+          .AppendLine(expectedXml.ToString())
+          .AppendLine()
+          .AppendLine("################### actual:")
+          .AppendLine(actualXml.ToString());
+
+        throw new Xunit.Sdk.XunitException(sb.ToString());
+      }
+    }
+
+    public static void AreJsonObjectsEqual(string expected, string actual)
+    {
+      var actualObject = JToken.Parse(actual);
+      var expectedObject = JToken.Parse(expected);
+      actualObject.Should().BeEquivalentTo(expectedObject);
+    }
+
+    public static void AreJsonObjectsSemanticallyEqual(string expected, string actual, IList<string>? excludePaths = null)
```

Comment:
excludePaths can be `IEnumerable<string>?`. I see no reason why we want to require the collection implementing `IList<string>` interface